src/presets/deno.ts

```diff
@@ -12,11 +18,119 @@ export const deno = defineNitroPreset({
   },
   rollupConfig: {
     preserveEntrySignatures: false,
-    external: ["https://deno.land/std/http/server.ts"],
+    // external: ["https://deno.land/std/http/server.ts"],
     output: {
       entryFileNames: "index.ts",
       manualChunks: () => "index",
       format: "esm",
     },
   },
 });
+
+export const denoServer = defineNitroPreset({
+  extends: "node-server",
+  entry: "#internal/nitro/entries/deno-server",
+  commands: {
+    preview:
+      "deno run --unstable --allow-net --allow-read --allow-env ./server/index.mjs",
+  },
+  rollupConfig: {
+    output: {
+      hoistTransitiveImports: false,
+    },
+    plugins: [
+      inject({
+        modules: {
+          process: "process",
+          global: "global",
+          Buffer: ["buffer", "Buffer"],
+          setTimeout: ["timers", "setTimeout"],
+          clearTimeout: ["timers", "clearTimeout"],
+          setInterval: ["timers", "setInterval"],
+          clearInterval: ["timers", "clearInterval"],
+          setImmediate: ["timers", "setImmediate"],
+          clearImmediate: ["timers", "clearImmediate"],
+        },
+      }),
+      {
+        name: "rollup-plugin-node-deno",
+        resolveId(id) {
+          id = id.replace("node:", "");
+          if (builtinModules.includes(id)) {
+            return {
+              id: `node:${id}`,
+              moduleSideEffects: false,
+              external: true,
+            };
+          }
+          if (isHTTPImport(id)) {
+            return {
+              id,
+              external: true,
+            };
+          }
+        },
+        renderChunk(code) {
+          const s = new MagicString(code);
+          const imports = findStaticImports(code);
+          for (const i of imports) {
+            if (
+              !i.specifier.startsWith(".") &&
+              !isAbsolute(i.specifier) &&
+              !isHTTPImport(i.specifier) &&
+              !i.specifier.startsWith("npm:")
+            ) {
+              const specifier = i.specifier.replace("node:", "");
+              s.replace(
+                i.code,
+                i.code.replace(
+                  new RegExp(`(?<quote>['"])${i.specifier}\\k<quote>`),
+                  JSON.stringify(
+                    builtinModules.includes(specifier)
+                      ? "node:" + specifier
+                      : "npm:" + specifier
+                  )
+                )
+              );
+            }
+          }
+          if (s.hasChanged()) {
+            return {
+              code: s.toString(),
+              map: s.generateMap({ includeContent: true }),
+            };
+          }
+        },
+      },
+      {
+        name: "inject-process",
```

Comment:
I'm wondering what is the cases that provide plugin fails to inject?

Reply:
it's not that it fails to inject, it's that the importMeta plugin adds it within `generateBundle` - which is too late for the inject plugin.